### ISSUE TITLE
blank login window with webkit-gtk-2.36.x

### ISSUE THREAD

**Issue by gagara** [dlenski/gp-saml-gui#41]
After upgrade to webkit-gtk-2.36.x login page is blank. Thus, not possible to enter credentials and proceed with login.

Downgrade to 2.34.x fixes the problem.

Distro: Gentoo

**Comment by jbulow** [comment]
Same problem on Ubuntu 20.04.

Workaround: Set WEBKIT_DISABLE_COMPOSITING_MODE=1 before starting gp-saml-gui.

The issue and workaround was found [here](https://bugs.launchpad.net/ubuntu/+source/evolution/+bug/1966418)

**Comment by DimitriPapadopoulos** [comment]
Isn't this a duplicate of #40?

**Comment by Zaijab** [comment]
I'm having the same issue. The workarounds in this thread are not working. The window seems to freeze and I must force quit. Here is what the terminal says when running,

```
zjabbar@euler ~/code/gp-saml-gui$ ./gp_saml_gui.py -S --gateway --clientos=Linux $SERVER
Looking for SAML auth tags in response to https://SERVER/ssl-vpn/prelogin.esp...
Got SAML POST, opening browser...
/home/zjabbar/code/gp-saml-gui/./gp_saml_gui.py:127: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed
  h = rs.get_http_headers() if rs else None
/home/zjabbar/code/gp-saml-gui/./gp_saml_gui.py:127: Warning: g_once_init_leave: assertion 'result != 0' failed
  h = rs.get_http_headers() if rs else None
Killed
zjabbar@euler ~/code/gp-saml-gui$ 
```

It used to work with the following packages in Guix
```
gtk+
webkitgtk-with-libsoup2
glib-networking
gsettings-desktop-schemas
python-pygobject
qtwebkit
network-manager-applet
hicolor-icon-theme
```

But now it does not anymore and I am quite confused.

**Comment by Zaijab** [comment]
Unfortunately not. It seems like that workaround is for Wayland users. I use X (EXWM in particular).

**Comment by DimitriPapadopoulos** [comment]
Then it's not the same issue, is it?

**Comment by Zaijab** [comment]
I would say it is not a duplicate issue. 

The warning the terminal gives (which is all I have to go on) is when the following is called: `rs.get_http_headers()` in line 114. Let me know if there are some more debugging techniques you would like to see me do. 

I am in no rush currently. I switched to my Windows OS which can access the VPN fine for my work.

**Comment by DimitriPapadopoulos** [comment]
If it's not the same issue, I recommend you open a new issue.

**Comment by Zaijab** [comment]
My issue is a blank login window appearing using `webkit-gtk-2.36.x`. This is the same issue as the current thread. I'm saying it is not a duplicate of #40. This issue is still open so I'm assuming it was not fixed yet.

**Comment by DimitriPapadopoulos** [comment]
It's as you wish. I was merely suggesting that we don't know for sure it's the same issue, and that adding new information that is potentially unrelated just make this issue more complex and less appealing to look into.

Also, you seem to suggest this is `webkit-gtk-2.36` issue.

Are you all on the same Linux distribution?

**Comment by DimitriPapadopoulos** [comment]
There are lots of issues all over internet about such Gnome issues:
https://www.google.com/search?q=WEBKIT_DISABLE_COMPOSITING_MODE+wayland+X+blank

I doubt it's worth opening issues against gp-saml-gui.

**Comment by Zaijab** [comment]
Makes sense. I'll work this issue out separately.

**Comment by jbicha** [comment]
Debian and Ubuntu are now using https://github.com/WebKit/WebKit/pull/18614 which I believe fixes the issue for most Nvidia users.
